src/bin/rustfmt.rs

```diff
@@ -263,7 +263,9 @@ fn main() {
 
     let exit_code = match execute(&opts) {
         Ok(summary) => {
-            if summary.has_operational_errors() {
+            if summary.diff_result > 0 {
```

Comment:
Could you add a comment (or even bettern assert, but I guess that is not possible) that this only happens in diff mode.